portfolio/src/main/java/com/google/sps/servlets/JournalServlet.java

```diff
@@ -0,0 +1,107 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Entity;
+import com.google.cloud.language.v1.Sentiment;
+import com.google.cloud.language.v1.Document;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.io.IOException;
+import com.google.sps.Sentence;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Query;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/journal")
+public class JournalServlet extends HttpServlet {
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    System.out.println("The journal input is "+request.getParameter("journal-input"));
+    // Get the input from the form.
+    String input = request.getParameter("journal-input");
+    long inputTime = System.currentTimeMillis();
+    int entrySize = input.length();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String[] entryAsStrings = input.split(";|\\.|\\?|\\!");
+    System.out.println("AS STRINGS "+entryAsStrings);
+    List<String> entryBySentence = new ArrayList<String>(Arrays.asList(entryAsStrings)); 
+    System.out.println("AS STRINGS IN LIST "+entryBySentence);
+
+    Collection sentences = new HashSet();
+    
+    int i = 0;
+    int avg = 0;
+    int wtAvg = 0;
+
+    System.out.println(entryBySentence.getClass().getName());
+    System.out.println(entryBySentence.get(i));
+
+    while (i<entryBySentence.size()) {
+        Sentence sentence = new Sentence(entryBySentence.get(i));
+        System.out.println(sentence.text());
+        System.out.println(sentence.score());
+        sentences.add(sentence);
+        i++;
+        avg += sentence.score();
+        wtAvg += sentence.score()*(sentence.text().length()/entrySize);
+    }
+
+    System.out.println("THE SENTENCES "+sentences);
+    System.out.println("WT AVG  "+wtAvg);
+     
+
+    Entity entryEntity = new Entity("Entry");
+    entryEntity.setProperty("content", sentences);
+    entryEntity.setProperty("timestamp", inputTime);
+    entryEntity.setProperty("average-score",avg/(entryBySentence.size()));
```

Comment:
What's the difference between "average-score" and "wtAvg" which you computed above? Why are you choosing one and not the other?

Reply:
explanation added.